### ISSUE TITLE
Incorrect conversion of IAM Offline Database to netcdf

### ISSUE THREAD

**Issue by alexde989** [cyber-tron/rnnlib-iam#2]
I am using this tool in order to convert the IAM Offline Dataset into a netCDF format. But when I use the converted netCDF in order to train the rnnlib I get the following errors-:

**training...
warning, seq lineImages/f01/f01-084/f01-084z-02.tif has requiredTime 26 > totalTime 15
warning, seq lineImages/p01/p01-130/p01-130z-06.tif has requiredTime 38 > totalTime 15
warning, seq lineImages/k09/k09-909/k09-909z-03.tif has requiredTime 41 > totalTime 16
warning, seq lineImages/b01/b01-074/b01-074z-08.tif has requiredTime 17 > totalTime 13
warning, seq lineImages/h07/h07-013/h07-013-05.tif has requiredTime 47 > totalTime 17
warning, seq lineImages/b01/b01-008/b01-008z-06.tif has requiredTime 37 > totalTime 16**

There is no handwriting recognition done. No output labels are produced, I am using the **offline.config** in your config folder.

I am uploading the entire log received from rnnlib.

Is this a configuration issue ? Or is off line conversion of IAM to netCDF not supported by this tool.

Are there any code examples using the IAM Offline Database ?

The online dataset seems to work well.

[log.txt](https://github.com/jan-auer/rnnlib-iam/files/171674/log.txt)

**Comment by ymnliu** [comment]
Hi @alexde989 , I encountered exactly the same issue. Have you solved this?

**Comment by soumayatou** [comment]
Hi all,
I have the same problem, and i don't find the solution. Can you help me please?

**Comment by akshaypardeshi26** [comment]
Has any of you solved this issue?
Can you help please?

**Comment by soumayatou** [comment]
You need to tune the hidden block and/input block parameters in your config file.

**Comment by akshaypardeshi26** [comment]
@soumayatou , Thanks. Can you be little more specific.
I tried different values for hidden block and input block. Training only works for block size (1,2) for all hidden and input blocks(in other cases I get warning). But in this case detected output sequence  is of length 0 or 1. 
Can you tell what exact parameters did you use in config file and did you make any other relative changes in any other file?

**Comment by soumayatou** [comment]
In my config file, I used this parameters:
hiddenSize 2,10,50
subsampleSize 6,20
hiddenBlock 2,4;1,4
inputBlock 3,4
learnRate 1e-4
momentum 0.9
optimiser steepest 
verbose false

If the warning was present yet, you have to resize your database.

**Comment by akshaypardeshi26** [comment]
I tried this configuration. Training is happening but everytime it detects output sequence of 0 or 1 characters.
Also, I tried changing ImageHeight and ImageWidth in iam_offline.py (resizing the database) then it takes higher block size as input too. But still my output sequence is of length 0 or 1(i checked for first 4-5 epochs).
Are you able to predict good output?

**Comment by samansarraf** [comment]
Hi guys, 
Is there any documentation describing the output of the toolbox? For example, what is the seqError and labelError? Are they training data error or testing data? How do need to perform testing or validation ? 
Regarding the issue you have reported , I am actually facing the same problem. I found an answer [here](https://sourceforge.net/p/rnnl/discussion/1183966/thread/fdb5a7d3/#e07f) . Does it help ?
Please advice,

**Comment by samansarraf** [comment]
@akshaypardeshi26 hi there, as @soumayatou mentioned above, you should play with input block parameters. Before seeing this message threads (I wish I saw it before), I changed 3,4 to 1,2 for my data and it decreased significantly the warnings. I would say tens of them turned into a few of them. If you read the paper, Alex Graves has discussed something about the block size and it seems the put this conditions in the code. Still, it's not fully clear how it works but I find out something will share with you.

**Comment by akash-flipkart** [comment]
@samansarraf You had any luck regarding the configuration for offline cases?

**Comment by samansarraf** [comment]
@akash-flipkart Actually , I did have some progress on my data not offline cases. Do you have any particular question?
